Autopilot/Src/reset_cause.c

```diff
@@ -0,0 +1,32 @@
+/*
+#define LPWRRSTF_Bit 31
+#define WWDGRSTF_Bit 30
+#define IWDGRSTF_Bit 29
+#define SFTRSTF_Bit 28
+#define PORRSTF_Bit 27
+#define PINRSTF_Bit 26 
+#define OBLRSTF_Bit 25 //Option Byte Loader
+*/
+
+#include "debug.h"
+#include "stm32f7xx_hal_rcc.h"
+/*
+void reset_cause_check(){
+    uint* RCC_CSR = 0x24;
+    if(*RCC_CSR & (1 << LPWRRSTF_Bit) == (1 << LPWRRSTF_Bit)){
+        debug("Lower power reset");
+    }
+    else if(*RCC_CSR & (1 << WWDGRSTF_Bit) == (1 << WWDGRSTF_Bit)){
+        debug("Window watchdog reset");
+    }
+
+*/
+
+void reset_cause_check(){
+    if(_HAL_RCC_GET_FLAG(RCC_FLAG_PINRST)) debug("Pin reset");
```

Comment:
None of this code compiles because this macro is typed wrong. Should be `__HAL_RCC_GET_FLAG()` (note the two leading underscores).